extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamInfoItemExtractor.java

```diff
@@ -194,11 +213,22 @@ public long getViewCount() throws ParsingException {
             if (videoInfo.getObject("topStandaloneBadge") != null || isPremium()) {
                 return -1;
             }
-            String viewCount = getTextFromObject(videoInfo.getObject("viewCountText"));
+
+            final JsonObject viewCountObject = videoInfo.getObject("viewCountText");
+            if (viewCountObject == null) {
+                // This object is null when a video has its views hidden.
+                return -1;
+            }
+
+            final String viewCount = getTextFromObject(viewCountObject);
+
+            if (viewCount.toLowerCase().contains("no views")) {
+                return 0;
```

Comment:
Are there no other ways than relying on English?

Reply:
At the moment no.

Reply:
Well there is almost always a way.
If it has BADGE_STYLE_TYPE_SIMPLE and can't extract views, it means it has no views.
For recommend, I tested with russian french german chinese thai, it stays « recommended for you »